### PR TITLE
Undersampling docs

### PR DESCRIPTION
**Pull request by jotaylor**
This PR updates the `undersampling_correction` step description to fix some grammatical typos and improve the readability. There are no associated issues or Jira issues. However, one thing I didn't know how to do was link to a subsection of another page. So the words "charge migration" in the 3rd sentence should link to the Charge Migration subsection of the ramp_fitting step: https://jwst-pipeline.readthedocs.io/en/latest/jwst/saturation/description.html#charge-migration
Can someone please tell me how to do so?

**Checklist for maintainers**
- [ ] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [ ] added relevant milestone
- [ ] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

File: docs/jwst/undersampling_correction/description.rst [hunk 1]
```diff
@@ -29,18 +29,15 @@ form of a `~jwst.datamodels.RampModel`.
...
-combining variances,
+--------- 
+The first group, and all subsequent groups, exceeding the value of the 
+``signal_threshold`` parameter is flagged as UNDERSAMP. ``signal_threshold`` is in units 
+of ADUs. These groups will also be flagged as DO_NOT_USE, and will not 
+be included in the slope calculation during the ``ramp_fitting`` step. Despite being flagged 
+as DO_NOT_USE, these UNDERSAMP groups are still included in the calculation of the readnoise. 
+This results in a readnoise variance for undersampled pixels that is similar to that of 
+pixels unaffected by charge migration. For the Poisson noise variance calculation in 
+``ramp_fitting``, the UNDERSAMP/DO_NOT_USE groups are not included.
```
Comment (hbushouse) [inline docs/jwst/undersampling_correction/description.rst:40]: again, could make this a link to the ramp_fitting docs

File: docs/jwst/undersampling_correction/description.rst [hunk 2]
```diff
@@ -49,3 +46,4 @@ Output product
 --------------
 The output is a new copy of the input `~jwst.datamodels.RampModel`, with the corrections applied
```
Comment (hbushouse) [inline docs/jwst/undersampling_correction/description.rst:47]: Hmmm, not sure I would characterize what this step does as "corrections". It's really just adding DQ flags, so that other steps later on do the right thing. So perhaps rephrase this to "with the updated DQ flags added to the GROUPDQ array."

**Review by hbushouse — COMMENTED** [review]
Looks good. Just a few minor suggestions.

Oh, you also need to add an entry to the CHANGES.rst (change log) file at the top level of the repo. But an entry in the "documentation" subsection and follow existing entries for the proper formatting, including the PR number (7589).

**Comment by jotaylor** [comment]
Thanks for the comments @hbushouse ! I think I've addressed them all, but please let me know if anything else needs updated.

File: docs/jwst/undersampling_correction/description.rst [hunk 3]
```diff
@@ -6,21 +6,22 @@ Description
...
+the spilling of charge from a saturated pixel into its neighboring pixels. This charge migration 
+causes group-to-group differences to decrease significantly once the signal level is greater than 
+~30,000 ADU. As a result, the last several groups of these ramps get flagged by the ``jump`` step. 
+The smaller number of groups used for these pixels in the ``ramp_fitting`` step results in them having 
+larger read noise variances, which in turn leads to lower weights used during resampling. This 
+ultimately leads to a lower than normal flux for the star in resampled images.
+
+Once a group in a ramp has been flagged as affected by charge migration, all subsequent 
+groups in the ramp are also flagged. By flagging these groups, they are not used in the 
+computation of slopes in the :ref:`ramp_fitting <ramp_fitting>` step. However, as described
```
Comment (nden) [inline docs/jwst/undersampling_correction/description.rst:22]: This reference is causing the build to fail. Where is the section located?

Comment (jotaylor) [reply docs/jwst/undersampling_correction/description.rst]: The name of the section was incorrect, I have fixed it

File: docs/jwst/undersampling_correction/description.rst [hunk 4]
```diff
@@ -6,21 +6,22 @@ Description
...
+the spilling of charge from a saturated pixel into its neighboring pixels. This charge migration 
+causes group-to-group differences to decrease significantly once the signal level is greater than 
+~30,000 ADU. As a result, the last several groups of these ramps get flagged by the ``jump`` step. 
+The smaller number of groups used for these pixels in the ``ramp_fitting`` step results in them having 
+larger read noise variances, which in turn leads to lower weights used during resampling. This 
+ultimately leads to a lower than normal flux for the star in resampled images.
+
+Once a group in a ramp has been flagged as affected by charge migration, all subsequent 
+groups in the ramp are also flagged. By flagging these groups, they are not used in the 
+computation of slopes in the :ref:`ramp_fitting <ramp_fit>` step. However, as described
```
Comment (hbushouse) [inline docs/jwst/undersampling_correction/description.rst:22]: This needs to be
```
:ref:`ramp_fitting <ramp_fitting_step>`
```
The first field after "ref" is the text that will actually appear in the doc, while the second field is the name of the reference/tag that needs to be linked, and the name of the tag (over in the ramp_fit step docs) is "ramp_fitting_step".

File: docs/jwst/undersampling_correction/description.rst [hunk 5]
```diff
@@ -29,23 +30,22 @@ form of a `~jwst.datamodels.RampModel`.
...
+--------- 
+The first group, and all subsequent groups, exceeding the value of the 
+``signal_threshold`` parameter is flagged as UNDERSAMP. ``signal_threshold`` is in units 
+of ADUs. These groups will also be flagged as DO_NOT_USE, and will not 
+be included in the slope calculation during the ``ramp_fitting`` step. Despite being flagged 
+as DO_NOT_USE, these UNDERSAMP groups are still included in the calculation of the
+variance due to readnoise. 
+This results in a readnoise variance for undersampled pixels that is similar to that of 
+pixels unaffected by charge migration. For the Poisson noise variance calculation in 
+:ref:`ramp_fitting <ramp_fit>`, the UNDERSAMP/DO_NOT_USE groups are not included.
```
Comment (hbushouse) [inline docs/jwst/undersampling_correction/description.rst:42]: same issue as above, where the second field (in angle brackets) needs to be "ramp_fitting_step".

**Review by hbushouse — COMMENTED** [review]
Couple of errors in syntax for the ramp_fit links need fixing.

File: docs/jwst/undersampling_correction/description.rst (hunk 4 above)
Comment (jotaylor) [reply docs/jwst/undersampling_correction/description.rst:22]: Gotcha. I was mistakenly looking at the `Alias` name in the ramp fitting description docs but I see the proper tag is in the index file. 👍

**Review by hbushouse — APPROVED** [review]
Looks good. All of the links now work correctly in the new docs.
